pkg/gitprovider/gitea/gitea.go

```diff
@@ -257,6 +276,56 @@ func (p *provider) ListPullRequests(
 	return prs, nil
 }
 
+// MergePullRequest implements gitprovider.Interface.
+func (p *provider) MergePullRequest(
+	ctx context.Context,
+	id int64,
+) (*gitprovider.PullRequest, bool, error) {
+	giteaPR, _, err := p.client.GetPullRequests(ctx, p.owner, p.repo, int(id))
+	if err != nil {
+		return nil, false, fmt.Errorf("error getting pull request %d: %w", id, err)
+	}
+	if giteaPR == nil {
+		return nil, false, fmt.Errorf("pull request %d not found", id)
+	}
+
+	switch {
+	case giteaPR.HasMerged:
+		pr := convertGiteaPR(*giteaPR)
+		return &pr, true, nil
+
+	case giteaPR.State != gitea.StateOpen:
+		return nil, false, nil
```

Comment:
For three out of five providers, you're treating a PR that hasn't been merged but is now in a non-open state as an error, which seems correct. Is there a reason that you're handling that differently here and for GitLab?